packages/office-ui-fabric-react/src/components/OverflowSet/OverflowSet.Props.ts

```diff
@@ -12,23 +16,40 @@ export interface IOverflowSetProps extends React.Props<OverflowSet> {
   componentRef?: (ref?: IOverflowSet) => void;
 
   /**
+   * Class name
+   */
+  className?: string;
+
+  /**
    * An array of items to be rendered by your onRenderItem function in the primary content area
   */
-  items?: any[] | IObjectWithKey[];
+  items?: IOverflowSetItemProps[];
 
   /**
    * An array of items to be passed to overflow contextual menu
   */
-  overflowItems?: any[];
+  overflowItems?: IOverflowSetItemProps[];
 
   /**
    * Method to call when trying to render an item.
   */
-  onRenderItem: IRenderFunction<any>;
+  onRenderItem: (item: IOverflowSetItemProps) => any;
 
   /**
    * Rendering method for overflow button and contextual menu. The argument to the function is
    * the overflowItems passed in as props to this function.
   */
   onRenderOverflowButton: IRenderFunction<any[]>;
 }
+
+export interface IOverflowSetItemProps {
+  /**
+   * Unique id to identify the item
+   */
+  key: string;
+
+  /**
+   * Any additional properties to use when custom rendering menu items.
+   */
```

Comment:
Generics are coming soon! David has a PR out, we should make sure to get rid of that index signature when that drops.